embulk-input-s3/src/main/java/org/embulk/input/s3/AbstractS3FileInputPlugin.java

```diff
@@ -67,6 +67,11 @@
         @ConfigDefault("null")
         public Optional<String> getAccessKeyId();
 
+        @Config("http_proxy")
+        @ConfigDefault("null")
+        public Optional<HttpProxy> getHttpProxy();
+        public void setHttpProxy(Optional<HttpProxy> httpProxy);
```

Comment:
Agreed. I changed my mind. Because my suggestion is not very good for this layer. I will introduce `env` option but, not on this PR. It will be implemented on https://github.com/embulk/embulk-util-config